### DISCUSSION TITLE
I'm trying to set theme for fish but it show some bugs

### DISCUSSION THREAD

**Discussion by datsfilipe** [fish-shell/fish-shell#9423] [Q&A] [answered]
I have this file called `theme.conf` that have this content:
```fish
# Set the theme variable to the desired theme
set -gx THEME gruvbox
```
And I have a bash script to update from gruvbox to another theme. In `config.fish`, I have this statement to source the theme file:
```fish
# Load theme configuration
source ~/.config/fish/theme.conf

# Load fish theme configuration
switch (echo $THEME)
  case "gruvbox"
    source (dirname (status --current-filename))/conf.d/gruvbox.fish
  case "tokyonight"
    source (dirname (status --current-filename))/conf.d/tokyonight.fish
end
```
Turns out that this happens:
![showing](https://user-images.githubusercontent.com/76636791/208809953-b0e9f2ec-35a0-4ace-9ece-9a16142ad7fe.gif)
I don't know what's happening. I'm using `tmux`, `fish` and `alacritty`.

**Comment by faho** [comment, marked as answer]
Is your gruvbox theme [this one](https://github.com/Jomik/fish-gruvbox)? It works by telling your terminal to change its palette colors. And it appears that your vim then sets the palette again, which would end up also changing it in fish. You would have to *reset* the palette after every command, e.g. by using the `fish_postexec` event:

```fish
function reset_theme --on-event fish_postexec
    # Load fish theme configuration
    switch "$THEME" # (your `echo` here was superfluous)
      case "gruvbox"
       source ~/.config/fish/conf.d/gruvbox.fish
      case "tokyonight"
       source ~/.config/fish/conf.d/tokyonight.fish
    end
end
```

All around this is an extremely fickle way of setting a colorscheme and not one I recommend. Do something like https://github.com/sujaybokil/fish-gruvbox and just set fish's color variables.

**Reply by datsfilipe** [reply]
Thanks, this solved for me, and no, not this one, I just created my own.
